presto-pinot/src/main/java/io/prestosql/pinot/PinotConnector.java

```diff
@@ -0,0 +1,107 @@
+import io.prestosql.spi.connector.ConnectorNodePartitioningProvider;
+import io.prestosql.spi.connector.ConnectorPageSourceProvider;
+import io.prestosql.spi.connector.ConnectorSplitManager;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.session.PropertyMetadata;
+import io.prestosql.spi.transaction.IsolationLevel;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class PinotConnector
+        implements Connector
+{
+    private static final Logger log = Logger.get(PinotConnector.class);
+    private final LifeCycleManager lifeCycleManager;
+    private final PinotMetadata metadata;
+    private final PinotSplitManager splitManager;
+    private final PinotPageSourceProvider pageSourceProvider;
+    private final PinotNodePartitioningProvider partitioningProvider;
+    private final List<PropertyMetadata<?>> sessionProperties;
+
+    @Inject
+    public PinotConnector(LifeCycleManager lifeCycleManager,
+                          PinotMetadata metadata,
+                          PinotSplitManager splitManager,
+                          PinotPageSourceProvider pageSourceProvider,
+                          PinotNodePartitioningProvider partitioningProvider,
+                          PinotSessionProperties pinotSessionProperties)
+    {
+        this.lifeCycleManager = requireNonNull(lifeCycleManager, "lifeCycleManager is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.splitManager = requireNonNull(splitManager, "splitManager is null");
+        this.pageSourceProvider = requireNonNull(pageSourceProvider, "pageSourceProvider is null");
+        this.partitioningProvider = requireNonNull(partitioningProvider, "partitioningProvider is null");
+        this.sessionProperties = ImmutableList.copyOf(requireNonNull(pinotSessionProperties, "sessionProperties is null").getSessionProperties());
+    }
+
+    @Override
+    public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly)
+    {
+        return PinotTransactionHandle.INSTANCE;
+    }
+
+    @Override
+    public ConnectorMetadata getMetadata(ConnectorTransactionHandle transactionHandle)
+    {
+        return metadata;
+    }
+
+    @Override
+    public ConnectorSplitManager getSplitManager()
+    {
+        return splitManager;
+    }
+
+    @Override
+    public ConnectorPageSourceProvider getPageSourceProvider()
+    {
+        return pageSourceProvider;
+    }
+
+    @Override
+    public ConnectorNodePartitioningProvider getNodePartitioningProvider()
+    {
+        return partitioningProvider;
+    }
+
+    @Override
+    public List<PropertyMetadata<?>> getSessionProperties()
+    {
+        return sessionProperties;
+    }
+
+    @Override
+    public final void shutdown()
+    {
+        try {
+            lifeCycleManager.stop();
+        }
+        catch (Exception e) {
```

Comment:
Remove the try-catch. The error is handled and reported by the caller (ConnectorManager).